pytorch_lightning/trainer/connectors/logger_connector.py

```diff
@@ -171,39 +173,40 @@ def _log_on_evaluation_epoch_end_metrics(self, epoch_logs):
         # --------------------------------
         # each dataloader aggregated metrics
         # now we log all of them
-        for dl_idx, dl_metrics in enumerate(step_metrics):
-            if len(dl_metrics) == 0:
-                continue
+        if reduce_on_epoch:
+            for dl_idx, dl_metrics in enumerate(step_metrics):
+                if len(dl_metrics) == 0:
+                    continue
 
-            reduced_epoch_metrics = dl_metrics[0].__class__.reduce_on_epoch_end(dl_metrics)
-            # make the keys 'k/dl'
-            reduced_epoch_metrics = self.__rename_keys_by_dataloader_idx(reduced_epoch_metrics, dl_idx, num_loaders)
+                reduced_epoch_metrics = dl_metrics[0].__class__.reduce_on_epoch_end(dl_metrics)
+                # make the keys 'k/dl'
+                reduced_epoch_metrics = self.__rename_keys_by_dataloader_idx(reduced_epoch_metrics, dl_idx, num_loaders)
 
-            # track the metrics
-            logger_metrics = reduced_epoch_metrics.get_epoch_log_metrics()
-            pbar_metrics = reduced_epoch_metrics.get_epoch_pbar_metrics()
-            self.logged_metrics.update(logger_metrics)
-            self.add_progress_bar_metrics(pbar_metrics)
+                # track the metrics
+                logger_metrics = reduced_epoch_metrics.get_epoch_log_metrics()
+                pbar_metrics = reduced_epoch_metrics.get_epoch_pbar_metrics()
+                self.logged_metrics.update(logger_metrics)
+                self.add_progress_bar_metrics(pbar_metrics)
 
-            # enable the metrics to be monitored
-            self.callback_metrics.update(logger_metrics)
-            self.callback_metrics.update(pbar_metrics)
+                # enable the metrics to be monitored
+                self.callback_metrics.update(logger_metrics)
+                self.callback_metrics.update(pbar_metrics)
 
-            # forked metrics were dropped, enable them for callbacks
-            forked_metrics = reduced_epoch_metrics.get_forked_metrics()
-            self.callback_metrics.update(forked_metrics)
+                # forked metrics were dropped, enable them for callbacks
+                forked_metrics = reduced_epoch_metrics.get_forked_metrics()
+                self.callback_metrics.update(forked_metrics)
 
-            # track the final results for the dataloader
-            self.eval_loop_results.append(deepcopy(self.callback_metrics))
+                # track the final results for the dataloader
+                self.eval_loop_results.append(deepcopy(self.callback_metrics))
 
-            # actually log
-            if len(logger_metrics) > 0:
-                metrics_to_log.append(logger_metrics)
+                # actually log
+                if len(logger_metrics) > 0:
+                    metrics_to_log.append(logger_metrics)
 
-        # log all the metrics as a s single dict
+    def log_on_evaluation_end(self, metrics_to_log):
```

Comment:
this needs to be more specific

log_on_evaluation_end (step or epoch?)

Reply:
Yes :) Changed it for `log_epoch_metrics_on_evaluation_end`.

Reply:
Please, switch to PR: https://github.com/PyTorchLightning/pytorch-lightning/pull/4215/files